cumulusci/core/source/github.py

```diff
@@ -146,16 +141,15 @@ def fetch(self, path=None):
                 shutil.rmtree(path)
                 raise
 
-        assert path.is_dir()
-
         project_config = self.project_config.construct_subproject_config(
             repo_info={
                 "root": os.path.realpath(path),
                 "owner": self.repo_owner,
                 "name": self.repo_name,
                 "url": self.url,
                 "commit": self.commit,
-            }
+            },
+            cache_dir=self.project_config.cache_dir,
```

Comment:
Should we put this in the implementation of `construct_subproject_config` so that all subprojects inherit the cache dir, or is there a reason we only want that to happen for subprojects constructed from a github source? The local folder source (folder.py) also calls construct_subproject_config

Reply:
THat's a good point. There's even an edge case where a local folder source could itself have a GitHub source that needs access to the parent cache.

Reply:
Done (plus test changes) in [58f733b](https://github.com/SFDO-Tooling/CumulusCI/pull/3092/commits/58f733b1f35bbc6a82f488639316104d5ccf9e14)